src/classes/BDI_DataImportService.cls

```diff
@@ -525,6 +529,20 @@ global with sharing class BDI_DataImportService {
                 populatedDataImportFields[i] = populatedDataImportFields[i].toLowerCase();
             }
 
+            this.flsValidator =
+                    new FieldLevelSecurityValidator(
+                            dataImports,
+                            fieldMapper.getTargetObjectsAndFieldsByDataImportField());
+
+            //From this point forward, only operate on the Data Import records that passed FLS
+            //validation.  Records that did not pass FLS validation will be updated with their
+            //respective FLS exceptions at the end of this block
+            if (isDryRun) {
+                this.listDI = flsValidator.getDataImportsWithAccessibleTargetFields();
```

Comment:
It's a bit wordy, but I love how descriptive these names are - there's a lot of complexity to FLS in this arena and it's worthwhile to be very clear about source vs target, accessible/createable/updateable, etc. Nice!